python/src/ops.cpp

```diff
@@ -2280,6 +2280,38 @@ void init_ops(py::module_& m) {
         Returns:
             list(array): A list of split arrays.
       )pbdoc");
+  m.def(
+      "fast_inference_sdpa",
+      &fast_inference_sdpa,
+      "q"_a,
+      "k"_a,
+      "v"_a,
+      "scale"_a,
+      "mask"_a = none,
+      py::kw_only(),
+      "stream"_a = none,
+      R"pbdoc(
+          fast_inference_sdpa(q: array, k: array, v: array,  /, *, stream: Union[None, Stream, Device] = None) -> array
+
+          A parallelized implementation of multi-head attention: O = softmax(Q @ K.T, dim=-1) @ V,
+          with a split along the KV sequence length.  Supports MHA, GQA, and MQA.
```

Comment:
```suggestion
          Scaled dot-product attention.
          
          Applies the function: ``O = softmax(Q @ K.T, dim=-1) @ V``
```

I didn't understand the comment `with a split along the KV sequence length`. That sounds like an implementation detail?

Also Regarding the different options `MHA`, `GQA` and `MQA`. I think those acronyms are not widely understood. Would be great to extend the docstring slightly to explain the constraints / supported cases.

Reply:
sgtm, will update and remove the comment regarding split along KV sequence length & expand MHA/GQA/MQA etc. with clarification on the tensor dimensions